addons/account/models/account_move.py

```diff
@@ -1302,6 +1303,69 @@ def _compute_quick_edit_mode(self):
             else:
                 move.quick_edit_mode = False
 
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    def _compute_duplicated_ref_ids(self):
+        move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
+        for move in self:
+            move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
+
+    def _fetch_duplicate_supplier_reference(self, only_posted=False):
+        moves = self.filtered(lambda m: m.is_purchase_document() and m.ref)
+        if not moves:
+            return {}
+
+        # this case deals with single move and thus avoid issue with those edited by UI
+        if len(moves) == 1:
+            duplicates = self.env["account.move"].search(
+                [
+                    ('company_id', '=', moves.company_id.id),
+                    ('move_type', '=', moves.move_type),
+                    ('commercial_partner_id', '=', moves.commercial_partner_id.id),
+                    ('ref', '=', moves.ref),
+                    ('state', '!=', 'cancel'),
+                ]
+                + ([
+                    ('invoice_date', '=', moves.invoice_date),
+                    ('state', '=', 'posted'),
+                ] if only_posted else [])
+                + ([('id', '!=', moves.ids[0])] if moves.ids else [])
```

Comment:
No need for the `.ids[0]`, `.id` is enough since you know that `len == 1`

Reply:
Coding something stupid for myself today, I come accros a situation in which `.id` was giving a in a compute `NewId` even when it is initiated from an already existing record.

Reply:
Interesting; today I learned.

For `.id`
https://github.com/odoo/odoo/blob/35805108a7e033a667694406e0f8c8b22bb17222/odoo/models.py#L6111-L6129
So this is the generic field's getter, taking values from the cache if it exists (and it does for new records)

And for `.ids`
https://github.com/odoo/odoo/blob/35805108a7e033a667694406e0f8c8b22bb17222/odoo/models.py#L5367-L5370
https://github.com/odoo/odoo/blob/35805108a7e033a667694406e0f8c8b22bb17222/odoo/models.py#L254-L258

So we only get "actual records", the NewId's are discarded if they don't correspond to an existing records.

Reply:
But looking for the origin explicitly makes it clearer though

Reply:
The case of `.id` remains quite unclear to me.